### PR TITLE
feat: add script to manually allocate for digest experiment

### PR DESCRIPTION
**Pull request by capJavert**
GB keeps variation assigned to users stable so with this script we pull variations for digest users and allocate manually so we can measure results. See [here](https://dailydotdev.slack.com/archives/C05G8BHHVBR/p1709038225690859?thread_ts=1709018432.641809&cid=C05G8BHHVBR)

When running I'll also check on my tests users first to verify it works.

### CODE REVIEW

File: bin/allocateDigestExperiment.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const con = await createOrGetConnection();
+
+  const personalizedDigestQuery = con
+    .createQueryBuilder()
+    .from(UserPersonalizedDigest, 'upd')
+    .where('upd."lastSendDate" > :lastSendDate', {
+      lastSendDate,
+    });
+
+  await loadFeatures(pino());
```
Comment (capJavert) [inline bin/allocateDigestExperiment.ts:36]: Load all features first.

File: bin/allocateDigestExperiment.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      await sendExperimentAllocationEvent(data);
+    },
+    allocationQueueConcurrency,
+  );
+
+  personalizedDigestStream.on(
+    'data',
+    (personalizedDigest: UserPersonalizedDigest) => {
+      const gbClient = getUserGrowthBookInstace(personalizedDigest.userId, {
+        trackingCallback: (experiment, result) => {
```
Comment (capJavert) [inline bin/allocateDigestExperiment.ts:54]: Get client for user id.

**Review by idoshamun — APPROVED** [review]
Amazing!

File: bin/allocateDigestExperiment.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  await loadFeatures(pino());
+
+  const personalizedDigestStream = await personalizedDigestQuery.stream();
+  const allocationQueueConcurrency = +(
+    process.env.ALLOCATION_QUEUE_CONCURRENCY || 1
+  );
+  const allocationQueue = fastq.promise(
+    async (data: ExperimentAllocationEvent) => {
+      // console.log('allocation', data);
```
Comment (rebelchris) [inline bin/allocateDigestExperiment.ts:44]: Console

**Review by rebelchris — APPROVED** [review]
Nice!

**Pull request merged**